src/api/spec/models/bs_request_spec.rb

```diff
@@ -220,6 +220,128 @@
       it { expect { request.change_state(newstate: 'review') }.to raise_error(PostRequestNoPermission) }
     end
 
+    context 'when the request is accepted' do
+      let(:creator) { create(:confirmed_user, login: 'sagan') }
+      let(:source_project) { create(:project, name: 'source_project_123', maintainer: creator) }
+      let(:source_package) { create(:package, project: source_project, name: 'source_package_123') }
+      let(:target_project) { create(:project, name: 'project_123', maintainer: creator) }
+      let(:target_package) { create(:package, project: target_project, name: 'package_123') }
+      let!(:request) do
+        create(:bs_request_with_submit_action, source_project: source_project, source_package: source_package, target_project: target_project, target_package: target_package,
+                                               creator: creator, description: 'A single comment')
+      end
+
+      context 'and the target project has an attribute to disallow acceptance by the creator' do
+        context 'and the accepter is the creator' do
+          before do
+            login creator
+
+            # Attach the attribute to the target project so to trigger the validation
+            attrib_type = AttribType.find_by_namespace_and_name!('OBS', 'CreatorCannotAcceptOwnRequests')
+            a = Attrib.create!(attrib_type: attrib_type, project: target_project)
+            a.values.create(value: '1')
+          end
+
+          it 'triggers an error' do
+            expect { request.change_state(newstate: 'accepted', force: true) }.to raise_error(BsRequest::Errors::CreatorCannotAcceptOwnRequests)
+          end
+        end
+
+        context 'and the accepter is NOT the creator' do
+          let(:user) { create(:confirmed_user, login: 'clarke') }
+
+          before do
+            # Make the user be part of the staff of the project, so we can send requests to it
+            create(:relationship_project_user, project: target_project, user: user)
+
+            login user
+
+            # Attach the attribute to the target project so to trigger the validation
+            attrib_type = AttribType.find_by_namespace_and_name!('OBS', 'CreatorCannotAcceptOwnRequests')
+            a = Attrib.create!(attrib_type: attrib_type, project: target_project)
+            a.values.create(value: '1')
+          end
+
+          it 'accepts the request' do
+            request.change_state(newstate: 'accepted', force: true)
+            expect(request.state).to be(:accepted)
+          end
+        end
+      end
+
+      context 'and the target project does not have an attribute to disallow acceptance by the creator' do
```

Comment:
This is already tested elsewhere I bet...

Reply:
Hm, okay, I'll remove those specs